src/util/boolset.rs

```diff
@@ -15,6 +15,7 @@ where
 {
     map: HashMap<K, bool>,
     default_val: bool,
+    pub is_default: bool,
```

Comment:
Why is this needed rather than something checking `self.map.is_empty()`? seems like that would avoid having to check for `is_default` before extending the map and so on.